---
- name: delugevpn - start
  community.docker.docker_container:
    name: delugevpn
    image: "{{ delugevpn_image }}"
    pull: true
    ports:
      - "8112:8112"
      - "8118:8118"
      - "58846:58846"
      - "58946:58946"
    env:
      VPN_ENABLED: "yes"
      VPN_USER: "{{ delugevpn_vpn_user }}"
      VPN_PASS: "{{ delugevpn_vpn_pass }}"
      VPN_PROV: "pia"
      VPN_CLIENT: "wireguard"
      STRICT_PORT_FORWARD: "yes"
      ENABLE_PRIVOXY: "yes"
      LAN_NETWORK: "192.168.0.0/16"
      NAME_SERVERS: "84.200.69.80,37.235.1.174,1.1.1.1,37.235.1.177,84.200.70.40,1.0.0.1"
      DELUGE_DAEMON_LOG_LEVEL: "warning"
      DELUGE_WEB_LOG_LEVEL: "warning"
      # VPN_INPUT_PORTS: "1234"
      # VPN_OUTPUT_PORTS: "5678"
      DEBUG: "false"
      UMASK: "022"
      PUID: "{{ puid | string }}"
      PGID: "{{ pgid | string }}"
    volumes:
      - "/etc/localtime:/etc/localtime:ro"
      - "delugevpn-data:/data"
      - "{{ app_path }}/delugevpn/config:/config"
      - "{{ app_path }}/data/downloads:/downloads"
    sysctls:
      net.ipv4.conf.all.src_valid_mark: 1
    privileged: true
    networks:
      - name: "{{ app_network_name }}"
    networks_cli_compatible: true
    labels:
      traefik.enable: "true"
      traefik.http.routers.delugevpn.rule: "HostRegexp(`deluge(.{{ inventory__homelab_domain }})?`)"
      traefik.http.services.delugevpn.loadbalancer.server.port: "8112"
      traefik.http.routers.delugevpn.entrypoints: "websecure"
      traefik.http.routers.delugevpn.tls: "true"
    restart_policy: unless-stopped
    state: started
  no_log: true
  tags: delugevpn
